src/encoder.ts

```diff
@@ -222,6 +222,30 @@ export class Encoder {
     };
   }
 
+  /**
+   * Returns an encoding function to be used on a message that is a date
+   */
+  static dateEncoder(minimum: number): EncodeFunc {
+    const f = Encoder.integerToPositiveInt(minimum);
+    return (v: unknown) => {
+      let dateVal = v instanceof Date ? v : null;
```

Comment:
if `v` is an instance of Date then next if block doesn't need to be considered. Just convert to int here.